zebra-consensus/src/transaction.rs

```diff
@@ -203,28 +240,13 @@ where
         let tx = request.transaction();
         let upgrade = request.upgrade(network);
 
-        // Do basic checks first
-        check::has_inputs_and_outputs(&tx)?;
-
-        // Handle transparent inputs and outputs.
-        if tx.is_coinbase() {
-            check::coinbase_tx_no_prevout_joinsplit_spend(&tx)?;
-        } else {
-            // feed all of the inputs to the script and shielded verifiers
-            // the script_verifier also checks transparent sighashes, using its own implementation
-            let cached_ffi_transaction = Arc::new(CachedFfiTransaction::new(tx.clone()));
-
-            for input_index in 0..inputs.len() {
-                let rsp = script_verifier.ready_and().await?.call(script::Request {
-                    upgrade,
-                    known_utxos: request.known_utxos(),
-                    cached_ffi_transaction: cached_ffi_transaction.clone(),
-                    input_index,
-                });
-
-                async_checks.push(rsp);
-            }
-        }
+        // Add asynchronous checks of the transparent inputs and outputs
+        async_checks.extend(Self::verify_transparent_inputs_and_outputs(
```

Comment:
Should the declaration above (`let async_checks = FuturesUnordered::new();`) be updated to be explicitly `AsyncChecks::new()`?

Reply:
Done, thanks!